### ISSUE TITLE
Allow additional selectors on :root

### ISSUE THREAD

**Issue by TazmanianD** [jhildenbiddle/css-vars-ponyfill#74]
This library currently requires the vars you define to be attached to a `:root` rule and exactly a `:root` rule. Previous issues have discussed allowing vars on other selectors and I understand why that's a challenge. I'm not sure I see why the rule has to be exactly `:root` though.

On our project, we're making use of a shadow DOM to render our app in browsers that support it and to do that our rule is `:root, :host`. When using a shadow DOM, we don't need to use your ponyfill so it's fine if this doesn't work for `:host`. But the problem is that the library won't even process these variables because of the extra `:host` selector.

This line of code seems to be the cause: https://github.com/jhildenbiddle/css-vars-ponyfill/blob/e22453fba24bdf0d593b0dbca17b05a947c7ed32/src/parse-vars.js#L40

Can that check be changed so that it only requires that one of the selectors be `:root` and can ignore additional selectors?

I think changing that to the following would do it unless there's some reason I'm not aware of that would make the library not work with the extra selector there.

```
if (rule.selectors.indexOf(":root") === -1)
```

**Comment by jhildenbiddle** [comment]
@TazmanianD --

Apologies for the delayed response. Was pulled away on other business for a few months, but I am now trying to catch up on GitHub issues.

TL;DR: Yep. I can make this change. Easy enough.

FWIW, the single `:root` selector limitation was put in place to enforce the limitations of the ponyfill and hopefully avoid confusion. For example, consider the following rulesets:

```css
:root {
  --color: green;
}

:root:hover {
  --color: red;
}
```

The above example doesn't seem like a common scenario, but who knows what folks are doing in their projects.

I'll add support for `:root` selectors in comma-separated selector lists as requested, as well as some copy in the docs to clarify usage.

**Comment by TazmanianD** [comment]
@simonwinklerzh This has been discussed before and you can find the previous discussion here: https://github.com/jhildenbiddle/css-vars-ponyfill/issues/26

**Comment by simonwinklerzh** [comment]
Hi @TazmanianD 

Thanks for the hint to #26 

However, I think that's a slightly different topic. #26 is about supporting variables defined in _any_ scope.

In my example, I'm still defining the variables on the :root element.
```
:root.theme-a {
	--main-color: green;
}

/* alternatively you could also write: */
.theme-a:root {
	--main-color: green;
}
```

Depending on which class is added to the <html> (root) element, the value of --main-color would be different.

Kind regards, Simon

**Comment by TazmanianD** [comment]
Oh yeah, you're right, my bad. Those selectors are still selecting the `:root` element but with extra "and" conditions to sometimes select them.

**Comment by jhildenbiddle** [comment]
@simonwinklerzh --

This could be done, but there would be a significant performance hit for doing so as the ponyfill would have to generate scoped CSS for each `:root` selector variation.

Consider the following:

```css
:root {
  --main-color: red;
}

:root.theme-a {
  --main-color: green;
}

:root.theme-b {
  --main-color: blue;
}

p {
  color: var(--main-color);
}
```

The ponyfill would have to generate the following CSS:

```css
p {
  color: red;
}

:root.theme-a p {
  color: green;
}

:root.theme-b p {
  color: blue;
}
```

Each CSS rule is duplicated for each `:root` variation. The obvious performance hit comes from asking the ponyfill to generate more CSS but--perhaps surprisingly--this isn't that big of an issue. The main concern is that the time it takes for the browser to parse large blocks of CSS and render the results is the most significant performance bottleneck, and the above approach would produce significantly more CSS for for the browser to parse.

The recommended way to swap themes is by using [`options.variables`](https://jhildenbiddle.github.io/css-vars-ponyfill/#/?id=variables) to swap values on demand.

```js
// Setup themes in JS
const themes = {
  a: {
    color: "green"
  },
  b: {
    color: "blue"
  }
};

// Initial run (default theme)
cssVars();

// Switch to theme "A"
cssVars({
  variables: themes.a
});

// Switch to theme "B"
cssVars({
  variables: themes.b
});
```

There are some advantages to keeping your theme variables in JS as well, such as the ability to quickly access these values from JS or perform calculations on them (such as auto-generating shades/hues).

Hope this helps!

**Comment by jhildenbiddle** [comment]
@TazmanianD --

This work is complete, btw. Just need to push a new version to npm. I'm on vacation currently, but I'll try to get a new version up in the next few days.

**Comment by simonwinklerzh** [comment]
@jhildenbiddle Sorry the late reply
Thanks for your insightful answer, it helped us making a decision.
Best Regards, Simon
